### PR TITLE
Issue #13797: Create a Checkstyle module to enforce comment formatting for input files

### LINKED ISSUE #13797: Create a Checkstyle module to enforce comment formatting in input files
We haveve noticed inconsistencies in the formatting of single-line comments in the codebase. For instance,  some comments lack a space after the //, making them harder to read. 

Current Behavior:
Comments like `//violation` exist in the codebase without a space after `//`.

Desired Behavior:
All single-line comments should have a space after `//`, like `// violation`.

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Resolves #13797

### CODE REVIEW

**Comment by relentless-pursuit** [comment]
I have just taken examples directory input files into consideration. Based on the review and feedback, we can expand the scope.

**Comment by romani** [comment]
Please create one more commit to cause violation and show that it works.

**Comment by relentless-pursuit** [comment]
> Please create one more commit to cause violation and show that it works.

There was logical error in the check. I have modified it and also deliberately modified one example file to be able to flag it. It worked in local. Will wait for the logs to fail.

File: config/checkstyle-examples-checks.xml [hunk 1]
```diff
@@ -82,12 +82,11 @@
   </module>
   <module name="RegexpSingleline">
     <property name="id" value="ensureSpaceAfterSpecificSingleLineComment"/>
-    <property name="format" value="^\s*//(ok|violation)[^ ]"/>
-    <property name="message" value="Comments like '//ok' or '//violation' should have a space after //."/>
+    <property name="format" value="^//(ok|violation)"/>
```
Comment (romani) [inline config/checkstyle-examples-checks.xml:85]: Please squash this change to first commit.
Test commit should be only input changes

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by romani** [comment]
Here is example  second reviewer https://ci.appveyor.com/project/Checkstyle/checkstyle/builds/48164885/job/cb577eql5bf1eki5#L123

Please remove second commit

**Comment by relentless-pursuit** [comment]
> Here is example second reviewer https://ci.appveyor.com/project/Checkstyle/checkstyle/builds/48164885/job/cb577eql5bf1eki5#L123
> 
> Please remove second commi

> Here is example second reviewer https://ci.appveyor.com/project/Checkstyle/checkstyle/builds/48164885/job/cb577eql5bf1eki5#L123
> 
> Please remove second commit

done

**Review by romani — APPROVED** [review]
Ok to merge

**Review by nrmancuso — APPROVED** [review]
Ok, it does not make sense to reuse module in `checkstyle-checks.xml`, since this limits us to only java files:

https://github.com/checkstyle/checkstyle/blob/58fdccf566a46b2ad4eea3987be3157d323bb3a8/config/checkstyle-checks.xml#L464-L471

We have other types of input that we might use this style of comment in, so let's keep this as a simple regexp match now and expand later as needed.

**Pull request merged**